### PR TITLE
Contract additions and `vmtools`

### PR DESCRIPTION
**Pull request by camilbancioiu**
This PR makes the following changes:
* Adds support for the `meta` crate in Rust contracts (with backwards compatibility)
* Running `erdpy deps install --overwrite all` will now focus strictly on known repositories and will not affect the Rust / golang installations (those can be updated individually)
* References the renamed VM repository throughout `erdpy` commands
* More migration from `str` to `pathlib.Path` where applicable
* Minor renaming

### CODE REVIEW

File: erdpy/testnet/setup.py [hunk 1]
```diff
@@ -246,34 +246,24 @@ def build_binaries(testnet_config: TestnetConfiguration):
...
-    logger.info(f"Arwen Binary: {arwen_binary}")
-    if arwen_binary:
-        logger.info("Building arwen...")
-        env = dict(golang_env)
-        env["ARWEN_PATH"] = str(node_folder)
-        node_folder_root = testnet_config.node_source()
-        myprocess.run_process(['make', 'arwen'], cwd=node_folder_root, env=env)
+    wasm_vm_binary = testnet_config.wasm_vm_binary()
+    if wasm_vm_binary:
+        logger.warn("WASM VM does not require building anymore. Skipping...")
```
Comment (andreibancioiu) [inline erdpy/testnet/setup.py:251]: :+1:

**Review by andreibancioiu — DISMISSED** [review]
Optionally, also run the help generation script (will update the CLI markdown file).

File: erdpy/projects/project_rust.py [hunk 2]
```diff
@@ -14,107 +14,132 @@
...
+        sink = myprocess.FileOutputSink(self.get_abi_filepath())
         return_code = myprocess.run_process_async(args, env=env, cwd=cwd, stdout_sink=sink)
         if return_code != 0:
             raise errors.BuildError(f"error code = {return_code}, see output")
 
-        utils.prettify_json_file(self._get_abi_filepath())
+        utils.prettify_json_file(self.get_abi_filepath())
+
+    def has_meta(self):
+        return self.get_meta_folder().exists()
```
Comment (andrei-marinica) [inline erdpy/projects/project_rust.py:104]:
```suggestion
        return (self.get_meta_folder() / "Cargo.toml").exists()
```

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: erdpy/projects/project_rust.py [hunk 3]
```diff
@@ -14,107 +14,132 @@
...
+        if return_code != 0:
+            raise errors.BuildError(f"error code = {return_code}, see output")
+
+    def run_meta(self):
+        cwd = self.get_meta_folder()
+        env = self.get_env()
+
+        args = [
+            "cargo",
+            "build",
```
Comment (andrei-marinica) [inline erdpy/projects/project_rust.py:72]:
```suggestion
            "run",
```

File: erdpy/projects/project_rust.py [hunk 4]
```diff
@@ -14,107 +14,132 @@
...
             env["RUSTFLAGS"] = "-C link-arg=-s"
 
-        cwd = path.join(self.directory, "wasm")
-        return_code = myprocess.run_process_async(args, env=env, cwd=cwd)
+        cwd = self.path / 'wasm'
+        return_code = myprocess.run_process_async(args, env=env, cwd=str(cwd))
+        if return_code != 0:
+            raise errors.BuildError(f"error code = {return_code}, see output")
+
+    def run_meta(self):
```
[the remark this answers is not in the document]
Comment (camilbancioiu) [reply erdpy/projects/project_rust.py:66]: Added.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by claudiu725 — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**